system-design/app-interoperability/workload-orchestration-agent.md

```diff
@@ -83,9 +83,214 @@ The following shall be configurable to ensure compliance with local operations.
+
+
+**Deployment specification example for Cluster with a single component**
+```
+apiVersion: margo.org/v1-alpha1
+kind: ApplicationDeployment
+metadata:
+    id: 0f83e682-a35b-4533-ae93-ff3810095f31
+    applicationId: com-northstarida-digitron-orchestrator
+spec:
+    deploymentProfile:
+      type: helm.v3
+      components:
+        - name: digitron-orchestrator
+          properties:
+            repository: oci://northstarida.azurecr.io/charts/northstarida-digitron-orchestrator
+            revision: 1.0.9
+    parameters:
+      hostname:
+        value: edge.host.local
+        targets:
+          - pointer: edge_host_name
+            components: ["digitron-orchestrator"]
+          - pointer: global.ingress.host
+            components: ["digitron-orchestrator", "database-services"]
+      pollFrequency:
+        value: 30
+        targets: 
+          - pointer: settings.pollFrequency
+            components: ["digitron-orchestrator"]
+      siteId:
+        value: d8b64e67-3bc4-4d0b-a3b8-da94b8f36315
+        targets:
+          - pointer: settings.siteId
+            components: ["digitron-orchestrator"]
+          - pointer: globals.site.id
+            components: ["database-services"]
+```
+
+**Top-level Attributes**
+
+| Attribute       | Type            | Required?       | Description     |
+|-----------------|-----------------|-----------------|-----------------|
+| apiVersion      | string    | Y    | Identifier of the version of the API the object definition follows.|
+| kind            | string    | Y    | Must be `ApplicationDeployment`.|
+| metadata        | Metadata    | Y    | Metadata element specifying characteristics about the application deployment. |
+| spec            | Spec    | Y    | Spec element that defines deployment profiles and parameters associated with the application deployment. |
+
+**Metadata Attributes**
+
+| Attribute        | Type            | Required?       | Description     |
+|-----------------|-----------------|-----------------|-----------------|
+| id             | string          | Y    | The unique identifier UUID of the deployment specification. Needs to be assigned by the Workload Orchestration Software.|
+| applicationID         | string          | Y    | An identifier for the application. The id is used to help create unique identifiers where required, such as namespaces. The id must be lower case letters and numbers and MAY contain dashes. Uppercase letters, underscores and periods MUST NOT be used. The id MUST NOT be more than 200 characters. |
+
+**Spec Attributes**
+
+| Attribute        | Type            | Required?       | Description     |
+|-----------------|-----------------|-----------------|-----------------|
+| deploymentProfile           | Target          | Y    | Section that defines which edge device the deployment specification is designed for|
+| parameters        | Parameters          | Y    | Describes the configured parameters applied via the end-user.|
```

Comment:
The type here should be a "map[string][Parameter]" - it's a dictionary of parameters using